test/common/http/http2/codec_impl_test.cc

```diff
@@ -502,6 +502,28 @@ TEST_P(Http2CodecImplFlowControlTest, EarlyResetRestoresWindow) {
   EXPECT_EQ(initial_connection_window, nghttp2_session_get_remote_window_size(client_.session()));
 }
 
+TEST_P(Http2CodecImplFlowControlTest, FlowControlPendingRecvData) {
```

Comment:
Can you add a one line `//` above explaining what is being validated? In general I prefer all tests to have this when non-trivial.